app/models/project.rb

```diff
@@ -8,25 +8,40 @@ class Project < ApplicationRecord
   scope :pending, -> { where(status: "pending") }
   scope :rejected, -> { where(status: "rejected") }
 
-  def approve
+  def approve(approval:)
     return true if approved?
-    update(status: "approved")
-  end
+    return false unless apply_status_change(approval, status: "approved")
 
-  def reject
-    return true if rejected?
-    update(status: "rejected")
+    members.each do |member|
+      ProjectStatusChangeMailer.approved(member: member, approval: approval).deliver
+    end
```

Comment:
It would be possible to move this into the `apply_status_change` method, but I wanted to separate the data mutation from the sending of the emails. I am also considering pulling the enqueuing of the emails to background job once we have background jobs setup.